api/types/header/header.go

```diff
@@ -0,0 +1,284 @@
+func FromMetadataV1(msg *headerv1.Metadata) *Metadata {
+	return &Metadata{
+		Name:        msg.Name,
+		Description: msg.Description,
+		Labels:      msg.Labels,
+		Expires:     msg.Expires.AsTime(),
+	}
+}
+
+// Metadata is resource metadata
+type Metadata struct {
+	// Name is an object name
+	Name string `json:"name" yaml:"name"`
+	// Description is object description
+	Description string `json:"description,omitempty" yaml:"description,omitempty"`
+	// Labels is a set of labels
+	Labels map[string]string `json:"labels,omitempty" yaml:"labels,omitempty"`
+	// Expires is a global expiry time header can be set on any resource in the system.
+	Expires time.Time `json:"expires" yaml:"expires"`
+	// ID is a record ID
+	ID int64 `json:"id,omitempty" yaml:"id,omitempty"`
+}
+
+// GetID returns the resource ID.
+func (m *Metadata) GetID() int64 {
+	return m.ID
+}
+
+// SetID sets the resource ID.
+func (m *Metadata) SetID(id int64) {
+	m.ID = id
+}
+
+// GetName returns the name of the resource.
+func (m *Metadata) GetName() string {
+	return m.Name
+}
+
+// SetName sets the name of the resource.
+func (m *Metadata) SetName(name string) {
+	m.Name = name
+}
+
+// SetExpiry sets the expiry time for the object.
+func (m *Metadata) SetExpiry(expires time.Time) {
+	m.Expires = expires
+}
+
+// Expiry returns the object expiry setting.
+func (m *Metadata) Expiry() time.Time {
+	return m.Expires
+}
+
+// Origin returns the origin value of the resource.
+func (m *Metadata) Origin() string {
+	if m.Labels == nil {
+		return ""
+	}
+	return m.Labels[common.OriginLabel]
+}
+
+// SetOrigin sets the origin value of the resource.
+func (m *Metadata) SetOrigin(origin string) {
+	if m.Labels == nil {
+		m.Labels = map[string]string{}
+	}
+	m.Labels[common.OriginLabel] = origin
+}
+
+// CheckAndSetDefaults verifies that the metadata object is valid.
+func (m *Metadata) CheckAndSetDefaults() error {
+	if m.Name == "" {
+		return trace.BadParameter("missing parameter Name")
+	}
+
+	// adjust expires time to UTC if it's set
+	if !m.Expires.IsZero() {
+		utils.UTC(&m.Expires)
+	}
+
+	for key := range m.Labels {
+		if !common.IsValidLabelKey(key) {
+			return trace.BadParameter("invalid label key: %q", key)
+		}
+	}
+
+	// Check the origin value.
+	if m.Origin() != "" {
+		if !slices.Contains(common.OriginValues, m.Origin()) {
+			return trace.BadParameter("invalid origin value %q, must be one of %v", m.Origin(), common.OriginValues)
+		}
+	}
+
+	return nil
+}
+
+// GetStaticLabels returns the static labels for the metadata.
+func (m *Metadata) GetStaticLabels() map[string]string {
+	return m.Labels
+}
+
+// SetStaticLabels sets the static labels for the metadata.
+func (m *Metadata) SetStaticLabels(sl map[string]string) {
+	m.Labels = sl
+}
+
+// GetLabel retrieves the label with the provided key. If not found
+// value will be empty and ok will be false.
+func (m *Metadata) GetLabel(key string) (value string, ok bool) {
+	v, ok := m.Labels[key]
+	return v, ok
```

Comment:
nit: Values are already named in the return values:
```suggestion
	return m.Labels[key]
```

Reply:
So this actually doesn't work for me:

```
# github.com/gravitational/teleport/api/types/header
api/types/header/header.go:277:9: not enough return values
	have (string)
	want (string, bool)
```

It looks like I need to do the get explicitly here.